libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -279,8 +279,7 @@ void ManagedQuery::submit_write(bool sort_coords) {
         query_->set_layout(
             sort_coords ? TILEDB_UNORDERED : TILEDB_GLOBAL_ORDER);
     }
-    query_->submit();
-    query_->finalize();
+    query_->submit_and_finalize();
```

Comment:
Only if not `sort_coords` IIUC

https://github.com/TileDB-Inc/TileDB/blob/2.23.0/tiledb/sm/query/query.cc#L449-L453